File: src/main/java/com/giant/auth/entity/AccountRole.java
package com.giant.auth.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Table(name = "account_role", schema = "auth")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class AccountRole {
    @Id
    @Column(name = "id", nullable = false)
    private Long accountRoleId;

    @Column(
            name = "role_name",
            nullable = false,
            unique = true
    )
    private String accountRoleName;

    @OneToMany(mappedBy = "accountRole", fetch = FetchType.LAZY)
    private List<Account> accounts;
}

File: src/main/kotlin/com/giant/auth/entity/AccountRole.kt
package com.giant.auth.entity

import jakarta.persistence.*

@Entity
@Table(name = "account_role", schema = "auth")
class AccountRole(
    @Id
    @Column(name = "id", nullable = false)
    var accountRoleId: Long = 0,

    @Column(
        name = "role_name",
        nullable = false,
        unique = true,
        length = 255,
        columnDefinition = "nvarchar(255) COLLATE Latin1_General_100_CI_AS_SC"
    )
    var accountRoleName: String = "",

    @OneToMany(mappedBy = "accountRole", fetch = FetchType.LAZY)
    val accounts: MutableList<Account> = mutableListOf()
)